### PR TITLE
CART-822 test: Add tracking of log line/mask/level frequency.

### PR DESCRIPTION
**Pull request by ashleypittman**
Print a report to stdout of how common various log lines and
facilities are used.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
Test failure is https://jira.hpdd.intel.com/browse/DAOS-4989

**Comment by ashleypittman** [comment]
This adds the ability to track which log locations and masks/debug levels are most used.  Currently only called from node_local_test but could be invoked for any log file.

File: src/cart/test/util/cart_logparse.py [hunk 1]
```diff
@@ -408,18 +414,28 @@ def __init__(self, fname):
...
-            data = self._fd.read(199)
-            lines = data.splitlines()
-            print(lines[-1])
-
-        self._fd.seek(0)
+
+        self.bz2 = False
+
+        if fname.endswith('.bz2'):
+            # Allow diret operation of bz2 files.  Supports multiple pids
```
Comment (brianjmurrell) [inline src/cart/test/util/cart_logparse.py:421]:
```suggestion
            # Allow direct operation of bz2 files.  Supports multiple pids
```

File: src/cart/test/util/cart_logtest.py [hunk 2]
```diff
@@ -233,6 +233,48 @@ def __init__(self, log_iter):
...
+                                            100*count/self.log_count))
+
+        print('Most common levels')
+        for (level, count) in self.log_levels.most_common(10):
+            if count < 10:
+                break
+            print('{}: {} ({:.1f}%)'.format(cart_logparse.LOG_NAMES[level],
+                                            count,
+                                            100*count/self.log_count))
+
```
Comment (brianjmurrell) [inline src/cart/test/util/cart_logtest.py:277]: Would this be better as an artifact?  Just to the goal of people not having to grub through logs.

Comment (ashleypittman) [reply src/cart/test/util/cart_logtest.py:277]: Yes, although there's the question of if you want one per log file, one per pid in each log file or one per entire build.

**Comment by ashleypittman** [comment]
> I found this after hunting:

> Can the log parser be run to only do this or does it automatically do all the checking? It would be nice to add this to other tests.

It can do either, it's our code.  If you run "cart_logtest.py filename" then it'll print anonymised messages at WARNING or higher, and the report you pasted here but we could change it to do something else if preferred.

As the next step I'd like this to be invoked by more tests as part of the post-analysis but that's a different PR.  If we're parsing the logfile it also wouldn't be too much work to make it do bzip2 compression on-the-fly if we wanted it it.

> Also, in the code, what you call "masks" is referred to as "facilities". Not sure that matters.

Will change.

**Comment by brianjmurrell** [comment]
> do bzip2 compression on-the-fly

This might be more difficult that it seems at first, because we actually use a parallel bzip2 compressor to take advantage of scads of cores.

**Pull request merged**